external/storm-hdfs/src/main/java/org/apache/storm/hdfs/spout/HDFSSpout.java

```diff
@@ -105,87 +105,87 @@
 
   private String configKey = Configs.DEFAULT_HDFS_CONFIG_KEY; // key for hdfs Kerberos configs
 
-  public HdfsSpout() {
+  public HDFSSpout() {
   }
 
-  public HdfsSpout setHdfsUri(String hdfsUri) {
+  public HDFSSpout setHdfsUri(String hdfsUri) {
     this.hdfsUri = hdfsUri;
     return this;
   }
 
-  public HdfsSpout setReaderType(String readerType) {
+  public HDFSSpout setReaderType(Class<? extends FileReader> readerType) {
```

Comment:
1. How about backward compatibility ? Changing the signature will break existing clients. May be you can deprecate the earlier method and add a new one like `setFileReader(Class<? extends FileReader> reader)`

2. How would one initialize a `HDFSSpout` via Flux? Earlier since the method accepted a String it was pretty trivial. Try it out and add some example, or add an additional method like `setFileReader(String className)` that takes the class name.